FirebaseRemoteConfig/Sources/RCNConfigRealtime.m

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2019 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import "FirebaseRemoteConfig/Sources/RCNConfigRealtime.h"
+#import <Foundation/Foundation.h>
+#import "FirebaseRemoteConfig/Sources/RCNConfigConstants.h"
+
+@implementation FIRConfigUpdateListenerRegistration {
+  RCNConfigRealtime *_realtimeClient;
+  FIRConfigUpdateCompletion completionHandler;
+}
+
+- (instancetype)initWithClient:(RCNConfigRealtime *)realtimeClient
+             completionHandler:(FIRConfigUpdateCompletion)completionHandler {
+  self = [super init];
+  if (self) {
+    _realtimeClient = realtimeClient;
+    completionHandler = completionHandler;
+  }
+  return self;
+}
+
+- (void)remove {
+  [self->_realtimeClient removeConfigUpdateListener:completionHandler];
+}
+
+@end
+
+@interface RCNConfigRealtime ()
+
+@property(strong, atomic, nonnull) NSMutableSet<FIRConfigUpdateCompletion> *listeners;
+@property(strong, atomic, nonnull) dispatch_queue_t realtimeLockQueue;
+
+@end
+
+@implementation RCNConfigRealtime
+
+- (instancetype)init {
+  self = [super init];
+  if (self) {
+    _listeners = [NSMutableSet alloc];
+    _realtimeLockQueue = [RCNConfigRealtime realtimeRemoteConfigSerialQueue];
+  }
+
+  return self;
+}
+
+/// Singleton instance of serial queue for queuing all incoming RC calls.
++ (dispatch_queue_t)realtimeRemoteConfigSerialQueue {
+  static dispatch_once_t onceToken;
+  static dispatch_queue_t realtimeRemoteConfigQueue;
+  dispatch_once(&onceToken, ^{
+    realtimeRemoteConfigQueue =
+        dispatch_queue_create(RCNRemoteConfigQueueLabel, DISPATCH_QUEUE_SERIAL);
+  });
+  return realtimeRemoteConfigQueue;
+}
+
+- (void)beginRealtimeStream {
+}
+
+- (void)pauseRealtimeStream {
+}
+
+- (FIRConfigUpdateListenerRegistration *)addConfigUpdateListener:
```

Comment:
i was under the impression that `addConfigUpdateListener` would only be called once, but here it looks like we have a set of listeners, so does that mean we will allow developers to register multiple listeners for the same config update signal?

Reply:
Yeah, we decided to allow for multiple listeners so that it would be easier to implement features in the wrapper languages like flows for Kotlin or publish in Swift. It's also not a lot of overhead to handle multiple listeners.

Reply:
I see. There's not much mention of this in the API proposal, maybe you could document this decision and mention how developers should handle the case where they have multiple `FIRConfigUpdateListenerRegistration`s